web-common/src/features/dashboards/pivot/pivot-utils.ts

```diff
@@ -51,6 +53,85 @@ export function getDimensionsInPivotColumns(
   );
 }
 
+export function getFormattedColumn(
+  pivot: PivotState,
+  allMeasures: MetricsViewSpecMeasureV2[],
+  alldimensions: MetricsViewSpecDimensionV2[],
+) {
+  const measures: string[] = [];
+  const dimensions: string[] = [];
+  const time: string[] = [];
+
+  const { columns } = pivot;
+
+  columns.forEach((colName) => {
+    const measure = allMeasures.find((m) => m?.name === colName);
+    const dimension = alldimensions.find((d) => d?.name === colName);
+
+    if (measure && measure.name) {
+      measures.push(measure.name);
+    } else if (dimension && dimension.name) {
+      dimensions.push(dimension.name);
+    } else {
+      time.push(colName);
+    }
+  });
+
+  return {
+    measures: measures.map((m) => {
+      return { id: m, title: m, type: PivotChipType.Measure };
+    }),
+    dimensions: dimensions.map((d) => {
+      return { id: d, title: d, type: PivotChipType.Dimension };
+    }),
+    time: time.map((t) => {
+      return { id: t, title: t, type: PivotChipType.Time };
```

Comment:
Lets use the `label` for titles here instead of `name` which is a dev identifier.